src/transformers/trainer.py

```diff
@@ -823,7 +833,21 @@ def train(self, model_path: Optional[str] = None, trial: Union["optuna.Trial", D
 
         return TrainOutput(self.state.global_step, tr_loss.item() / self.state.global_step)
 
-    def _maybe_log_save_evalute(self, tr_loss, model, trial, epoch):
+    def _rewrite_metrics(self, prefix, metrics):
```

Comment:
This does not sound very good. Lots of behaviors and tests rely on the metric names, so I'd avoid any renaming in the main Trainer.

Reply:
I did not change the eval part, for this reason. But I was not aware of same issues with train metrics. I can move code, but I will have to change callbacks to send more information to do the same job in the integrations (not really a problem of course).
Another way would be to change the tests, as prefixing metrics with eval_ is not really easy to handle, as metrics name themselves may contain underscores, whereas splitting by / would make it easier to handle different metrics.
But as you said, if a lot of behaviors are relying on this names, and not just tests, it's hard to change everywhere.
The eternal problem between evolving code and keeping backward-compatibility...

Reply:
Fixed in https://github.com/huggingface/transformers/pull/7935/commits/d57d45fe6246e4d6f85a7dc683858546da569e18